File: app/src/main/java/com/example/projectbank/Interface/UserService.java
package com.example.projectbank.Interface;

import com.example.projectbank.Model.User;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface UserService {

    @FormUrlEncoded
    @POST("login")
    Call<User> buscarUser(@Field("user") String user, @Field("password") String password);
}

File: SantanderApp/app/src/main/java/com/qintess/santanderapp/service/UserService.kt
package com.qintess.santanderapp.service

import com.qintess.santanderapp.model.CredentialsModel
import com.qintess.santanderapp.model.UserModel

open class UserService: ServiceInterface {

    fun login(credentials: CredentialsModel, onSuccess: SuccessCallback<UserModel>, onFailure: FailureCallback) {
        val bodyParameters = RequestParameters()
        bodyParameters["user"] = credentials.user
        bodyParameters["password"] = credentials.password

        getHttpClient().post(
            "login", bodyParameters,
            onSuccess = { responseJsonObj ->
                val userJsonObj = responseJsonObj.getJSONObject("userAccount")
                val user = UserModel(
                    userId = userJsonObj.getInt("userId"),
                    name = userJsonObj.getString("name"),
                    bankAccount = userJsonObj.getString("bankAccount"),
                    agency = userJsonObj.getString("agency"),
                    balance = userJsonObj.getDouble("balance")
                )
                onSuccess(user)
            },
            onFailure = onFailure
        )
    }

    override fun getHttpClient(): HttpInterface {
        return Http()
    }
}